src/ar_gateway_h.erl

```diff
@@ -84,6 +58,50 @@ handle_labeled_request_2(Domain, Label, TXID, Filename, BH, Req) ->
 		OtherLabel -> redirect_to_labeled_tx(Domain, OtherLabel, TXID, Req)
 	end.
 
+handle_custom_request(Domain, CustomDomain, Req) ->
```

Comment:
The amount of details accounted for in this module is not trivial, I think it would be very helpful to add a generic comment to each of `handle_custom_request`, `handle_labeled_request` and `handle_apex_request`. For `handle_custom_request` I see something along the lines of:

```
handle_custom_domain ensures that:

- "mycustomdomain.site" serves the transaction specified in its txt record.
  It has to be passed on  upon startup as "custom_domain".
- "mycustomdomain.site/[TXID]" redirects to "[TXPREFIX].gateway.com/[TXID]" where "gateway.com" is the domain mycustomdomain.site points to.
```